### PR TITLE
Remove stream from SendMagicLink

### PR DESCRIPTION
**Pull request by lambdaR**
Now, SendMagicLink does the necessary checks and creates a cache for the token then sends the MgicLink to the specified email.

VerifyToken does the necessary checks and then sends back a session object if all the checks are passed.

### CODE REVIEW

**Comment by lambdaR** [comment]
Is it ok with this email address here https://github.com/micro/services/blob/3cdf5baa23a82adc188764e10c621fbeaa229ee9/user/domain/domain.go#L487 for the From part?

File: user/proto/user.proto [hunk 1]
```diff
@@ -272,5 +270,6 @@ message VerifyTokenRequest {
 
 message VerifyTokenResponse {
 	bool is_valid = 1;
-	string message = 2;
+	Session sessino = 2;
```
Comment (asim) [inline user/proto/user.proto:273]: Typo in session

File: user/examples.json [hunk 2]
```diff
@@ -258,7 +251,8 @@
       },
       "response": {
         "is_valid": false,
-        "message": ""
+        "message": "",
+        "session": {}
```
Comment (asim) [inline user/examples.json:255]: Can you provide a dummy session here

File: user/proto/user.proto (hunk 1 above)
Comment (lambdaR) [reply user/proto/user.proto]: done

File: user/examples.json (hunk 2 above)
Comment (lambdaR) [reply user/examples.json]: done

**Comment by lambdaR** [comment]
@asim SendMagicLink returns `{"Id":"user","Code":500,"Detail":"panic recovered: runtime error: invalid memory address or nil pointer dereference","Status":"Internal Server Error"}` i suspect that this line creates a panic some how maybe `users[0].Username` https://github.com/micro/services/blob/2b24aba3c3163ce992cbb175d2bbfa6f713518b1/user/handler/handler.go#L426 can you check the trace in the log?

**Comment by asim** [comment]
```
21-12-13 14:32:16  file=grpc/grpc.go:231 level=error panic recovered: runtime error: invalid memory address or nil pointer dereference
2021-12-13 14:32:16  file=grpc/grpc.go:232 level=error goroutine 1761 [running]:
runtime/debug.Stack(0xc00000e5a0, 0xc0007bca02, 0xc0007bca60)
        /usr/local/go/src/runtime/debug/stack.go:24 +0x9f
github.com/micro/micro/v3/service/server/grpc.(*grpcServer).handler.func1(0xc00023d380, 0xc000577c08)
        /root/go/pkg/mod/github.com/micro/micro/v3@v3.8.0/service/server/grpc/grpc.go:232 +0x437
panic(0xde9b00, 0x15c1a70)
        /usr/local/go/src/runtime/panic.go:975 +0x47a
github.com/micro/services/user/handler.(*User).SendMagicLink(0xc00010e360, 0x104e6c0, 0xc0004993e0, 0xc0004e4900, 0xc000572780, 0x4ccf25, 0xe3fd00)
        /tmp/src070548220/user/handler/handler.go:407 +0x119
```

**Pull request merged**